### PR TITLE
Re-structure lint tasks

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Adds named functions for lint tasks instead of anonymous functions.
- Fixes incorrect path for build files in lint task.

## Testing

- `gulp lint` should pass.

## Notes

- Spun out of https://github.com/cfpb/capital-framework/pull/726

### CODE REVIEW

File: scripts/gulp/lint.js [hunk 1]
```diff
@@ -22,53 +20,63 @@ function _genericLintJs( src ) {
...
 /**
  * Lints the test js files for errors.
+ * @returns {Object} An output stream from gulp.
  */
-gulp.task( 'lint:tests', () => {
-  _genericLintJs( [
+function lintTests() {
+  return _genericLintJs( [
     'test/accessibility/*.js',
     'test/*.js'
```
Comment (jimmynotjim) [inline scripts/gulp/lint.js]: This could be simplified as `test/**/*.js`. Currently it's not catching any of the unit tests for atomic component

**Comment by Scotchester** [comment]
Probably a dumb question, but: Why?

**Comment by anselmbradford** [comment]
> Probably a dumb question, but: Why?

Named functions are generally better as they'll have a name in any call stack errors. It also allows us to group the task definitions together at the bottom, and it's the [gulp recommend approach](https://github.com/gulpjs/gulp/issues/802#issuecomment-66195466). Lastly, ESlint will complain about undocumented functions, but it won't complain about undocumented anonymous functions.

File: scripts/gulp/lint.js (hunk 1 above)
Comment (anselmbradford) [reply scripts/gulp/lint.js]: Done! Good catch

**Pull request merged**